### PR TITLE
feat: add auto refresh to notebooks

### LINKED ISSUE #3532: Add Autorefresh for Notebooks
As notebooks starts to gain traction as the main part of our application, we want to expand the existing functionality so that transitioning from other parts of the app won't mean sacrificing out on features. In addition, with the current work that's being done for error threshold triggering on the UI, we're going to need a way to run queries in the background (without a task) for individuals who want to present live data in presentation mode over large screens. We should use the existing design and infrastructure to expand the existing auto refresh capabilities into notebooks.

Current Autorefresh Design:
Screen Shot 2021-12-02 at 12 16 1
Screen Shot 2021-12-02 at 12 17 1 (2)

New Autorefresh Design:
[Figma](https://www.figma.com/file/qP1atZtFp9axPu8BRYuvQk/Dashboards-Enable-Autorefresh?node-id=0%3A1)
Autorefresh _ Indefinite
Autorefresh _ Custom (1)

### PR DESCRIPTION
**Pull request by asalem1**
Closes #3532

This PR addresses a gap in functionality between the dashboards and notebooks. By extending the Auto Refresh function onto notebooks from dashboards, we can easily transition existing dashboards with the functionality into the UI. In addition, this feature allows us to have view-oriented functionality that's triggered on query runs.

### Product Question

Do we want to make this available for shared notebooks? What's the expectation around a user that's shared a notebook with an autorefresh set? Do we expect that state to persist across browsers or only locally?

autorefresh

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 51 | :red_circle:
accessibility | 97 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=3646)

File: src/flows/context/flow.current.tsx [hunk 1]
```diff
@@ -54,6 +62,79 @@ export const FlowProvider: FC = ({children}) => {
     }
   }
 
+  const {autoRefresh} = useSelector((state: AppState) => {
```
Comment (drdelambre) [inline src/flows/context/flow.current.tsx:65]: this stuff shouldnt be in here, but in it's own component

**Review by drdelambre — CHANGES_REQUESTED** [review]
code looks great, it's just in the wrong place

File: src/flows/context/flow.query.tsx [hunk 2]
```diff
@@ -334,6 +335,20 @@ export const FlowQueryProvider: FC = ({children}) => {
     _queryAll()
   }
 
+  const handleAutoRefresh = () => {
```
[the remark this answers is not in the document]
Comment (drdelambre) [reply src/flows/context/flow.query.tsx:338]: move this to the button you made

[the remark this answers is not in the document]
Comment (asalem1) [reply src/flows/context/flow.query.tsx:338]: aw whoops

**Review by drdelambre — APPROVED** [review]

**Pull request merged**